assets/helpers/flashSale.js

```diff
@@ -1,84 +1,225 @@
+          },
+          landingPage: {
+            heading: 'Digital Pack',
+            subHeading: 'Save 25% - $16.13/month for a year, then $21.50/month',
+          },
+          bundle: {
+            heading: 'Digital Pack',
+            subHeading: 'Save 25% for a year',
+            description: 'The Premium App and the daily edition iPad app of the UK newspaper in one pack, plus ad-free reading on all your devices',
+          },
+        },
+      },
+
+    },
+  },
+];
+
+function sortSalesByStartTimesDescending(a: Sale, b: Sale) {
+  return b.startTime - a.startTime;
+}
+
+function getSales(product: SubscriptionProduct, countryGroupId: CountryGroupId = detect()): Sale[] {
+  return Sales.filter(sale =>
+    sale.subscriptionProduct === product &&
+    sale.activeRegions.includes(countryGroupId)).sort(sortSalesByStartTimesDescending);
+}
+
+function flashSaleIsActive(product: SubscriptionProduct, countryGroupId: CountryGroupId = detect()): boolean {
+  const now = Date.now();
+  const sales = getSales(product, countryGroupId).filter(sale =>
+    (now > sale.startTime && now < sale.endTime) ||
+    getQueryParameter('flash_sale') === 'true');
+
+  return sales.length > 0;
+}
+
+function getPromoCode(
+  product: SubscriptionProduct,
+  countryGroupId?: CountryGroupId = detect(),
+  defaultCode: string,
+): string {
+  if (flashSaleIsActive(product, countryGroupId)) {
+    const sale = getSales(product, countryGroupId)[0];
+    return sale && sale.saleDetails[countryGroupId].promoCode;
   }
   return defaultCode;
 }
 
-function getIntcmp(product: SubscriptionProduct, intcmp: ?string, defaultIntcmp: string): string {
+function getIntcmp(
+  product: SubscriptionProduct,
+  countryGroupId: CountryGroupId = detect(),
+  intcmp: ?string,
+  defaultIntcmp: string,
+): string {
   if (flashSaleIsActive(product)) {
-    return saleDetails[product].intcmp || intcmp || defaultIntcmp;
+    const sale = getSales(product, countryGroupId)[0];
+    return (sale && sale.saleDetails[countryGroupId].intcmp) || intcmp || defaultIntcmp;
   }
   return intcmp || defaultIntcmp;
 }
 
+function getEndTime(product: SubscriptionProduct, countryGroupId: CountryGroupId) {
+  const sale = getSales(product, countryGroupId)[0];
+  return sale && sale.endTime;
+}
+
+function getSaleCopy(product: SubscriptionProduct, countryGroupId: CountryGroupId): SaleCopy {
+  const emptyCopy = {
+    featuredProduct: {
+      heading: '',
+      subHeading: '',
+      description: '',
+    },
+    landingPage: {
+      heading: '',
+      subHeading: '',
+    },
+    bundle: {
+      heading: '',
+      subHeading: '',
+      description: '',
+    },
+  };
+  if (flashSaleIsActive(product, countryGroupId)) {
+    const sale = getSales(product, countryGroupId)[0];
+    return sale.saleDetails[countryGroupId].saleCopy;
+  }
+  return emptyCopy;
+}
+
+function getFormattedFlashSalePrice(product: SubscriptionProduct, countryGroupId: CountryGroupId): string {
```

Comment:
Is `discountedDisplayPrice()` in subscriptions.js unused now this function exists? If so you could remove it.

Reply:
It's unused from the POV of flash sales. I'll check to see if that was the only invocation. Thanks.

Reply:
Turns out there were a few related things that could also go. So they're gone.
Re-tested (locally) - all ok.